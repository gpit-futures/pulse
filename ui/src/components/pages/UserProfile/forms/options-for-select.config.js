const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

const optionsForCountryField = [{
  value: 'usa',
  title: 'USA',
}, {
  value: 'country1',
  title: 'Country 1',
}, {
  value: 'country2',
  title: 'Country 2',
}, {
  value: 'country3',
  title: 'Country 3',
}, {
  value: 'country4',
  title: 'Country 4',
}, {
  value: 'country5',
  title: 'Country 5',
}];

export { optionsForThemesField, optionsForGenderField, optionsForCountryField }
