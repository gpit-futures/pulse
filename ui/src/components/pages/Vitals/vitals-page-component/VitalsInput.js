import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import _ from 'lodash/fp';

import VitalsPopover from './VitalsPopover'

export default class VitalsInput extends PureComponent {
  static propTypes = {
    label: PropTypes.string.isRequired,
    labelCheckbox: PropTypes.string,
    placeholder: PropTypes.string,
    input: PropTypes.object.isRequired,
    meta: PropTypes.shape({
      active: PropTypes.bool,
      error: PropTypes.any,
    }).isRequired,
  };

  state = {
    isChanged: false,
  };

  componentWillReceiveProps(nextProps) {
    if (nextProps.meta.dirty) {
      this.setState({ isChanged: true })
    }
  }

  render() {
    const { addonName, label, placeholder, input, type, meta: { error, touched }, id, disabled, isSubmit, isNotValidate, getHighlighterClass, popoverLabels, vitalStatuses, detail, isInput, withoutPopover } = this.props;
    const { isChanged } = this.state;
    const showError = ((touched || isChanged || isSubmit) && error);
    return (
      <div className={classNames('vitals-group highlighter-wrapper', { 'has-error': showError && !isNotValidate }, { 'has-success': isChanged && !error && !isNotValidate })}>
        <span className={`${getHighlighterClass(input.name)}`} />
        <label className="vitals-label">{label}</label>
        {!withoutPopover ? <VitalsPopover
          title={label}
          popoverLabels={(!_.isEmpty(popoverLabels[input.name])) ? popoverLabels[input.name] : []}
          vitalStatusesType={(!_.isEmpty(vitalStatuses[input.name])) ? vitalStatuses[input.name].type : ''}
          detailValue={detail[input.name]}
          vitalsAddon={addonName}
          isInput={isInput}
          placeholder={placeholder}
          id={id}
          type={type}
          disabled={disabled}
          input={input}
          error={error}
          showError={showError}
          isNotValidate={isNotValidate}
        /> :
          <div className={`input-holder vitals-holder ${vitalStatuses[input.name].type}`}>
            <input
              className="form-control input-sm"
              placeholder={placeholder}
              id={id}
              type={type}
              disabled={disabled}
              {...input}
            />
          </div>}
      </div>
    )
  }
}
