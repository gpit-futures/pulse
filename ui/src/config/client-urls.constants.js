import { themeConfigs } from '../themes.config';

export const clientUrls = {
  ROOT: '/',
  CHARTS: '/charts',
  PATIENTS: '/patients',
  PATIENTS_FULL_DETAILS: '/patients-full-details',
  SEARCH_REPORT: '/search-report',
  USER_PROFILE: '/profile',
  PATIENTS_SUMMARY: 'patients-summary',
  RESULTS: 'results',
  DIAGNOSES: 'diagnoses',
  ALLERGIES: 'allergies',
  CONTACTS: 'contacts',
  CLINICAL_NOTES: 'clinicalNotes',
  PERSONAL_NOTES: 'personalNotes',
  VACCINATIONS: 'vaccinations',
  MEDICATIONS: 'medications',
  GENERIC_PLUGIN: 'genericPlugin',
  PROCEDURES: 'procedures',
  EVENTS: 'events',
  TEST_RESULTS: 'results',
  REFERRALS: 'referrals',
  ORDERS: 'orders',
  MDTS: 'mdt',
  DRAWINGS: 'drawings',
  VITALS: 'vitals',
  DOCUMENTS: 'documents',
  IMAGES: 'images',
  TRANSFERS_OF_CARE: 'transfer-of-care',
  TOP_THREE_THINGS: 'topThreeThings',
  CLINICAL_STATEMENTS: 'clinicalStatements',
  PROMS: 'proms',
  FEEDS: 'feeds',
};

const pluginsPages = {
  'allergies': {
    breadcrumbs: [{
      title: 'Allergies',
      state: '/allergies',
    }],
  },
  'diagnoses': {
    breadcrumbs: [{
      title: 'Problems / Diagnoses',
      state: '/diagnoses',
    }],
  },
  'clinicalNotes': {
    breadcrumbs: [{
      title: 'Clinical Notes',
      state: '/clinicalNotes',
    }],
  },
  'personalNotes': {
    breadcrumbs: [{
      title: 'Personal Notes',
      state: '/personalNotes',
    }],
  },
  'contacts': {
    breadcrumbs: [{
      title: 'Contacts',
      state: '/contacts',
    }],
  },
  'vaccinations': {
    breadcrumbs: [{
      title: 'Vaccinations',
      state: '/vaccinations',
    }],
  },
  'genericPlugin': {
    breadcrumbs: [{
      title: 'Generic Plugin',
      state: '/genericPlugin',
    }],
  },
  'medications': {
    breadcrumbs: [{
      title: 'Medications',
      state: '/medications',
    }],
  },
  'procedures': {
    breadcrumbs: [{
      title: 'Procedures',
      state: '/procedures',
    }],
  },
  'events': {
    breadcrumbs: [{
      title: 'Events',
      state: '/events',
    }],
  },
  'results': {
    breadcrumbs: [{
      title: 'Test Results',
      state: '/results',
    }],
  },
  'referrals': {
    breadcrumbs: [{
      title: 'Referrals',
      state: '/referrals',
    }],
  },
  'orders': {
    breadcrumbs: [{
      title: 'Orders',
      state: '/orders',
    }],
  },
  'mdt': {
    breadcrumbs: [{
      title: 'MDT',
      state: '/mdt',
    }],
  },
  'drawings': {
    breadcrumbs: [{
      title: 'Drawings',
      state: '/drawings',
    }],
  },
  'vitals': {
    breadcrumbs: [{
      title: 'Vitals - NEWS',
      state: '/vitals',
    }],
  },
  'documents': {
    breadcrumbs: [{
      title: 'Documents',
      state: '/documents',
    }],
  },
  'images': {
    breadcrumbs: [{
      title: 'Images',
      state: '/images',
    }],
  },
  'transfer-of-care': {
    breadcrumbs: [{
      title: 'Transfers Of Care',
      state: '/transfers-of-care',
    }],
  },
  'topThreeThings': {
    breadcrumbs: [{
      title: 'Top 3 Things',
      state: '/topThreeThings',
    }],
  },
  'clinicalStatements': {
    breadcrumbs: [{
      title: 'Clinical Statements',
      state: '/clinicalStatements',
    }],
  },
  'proms': {
    breadcrumbs: [{
      title: 'PROMs',
      state: '/proms',
    }],
  },
};


const addPluginsPagesToLists = (list, breadcrumbsBefore) => {
  for (const nameOfPage in pluginsPages) {
    const breadcrumbs = breadcrumbsBefore.concat(pluginsPages[nameOfPage].breadcrumbs);
    list[nameOfPage] = { breadcrumbs };
  }
};

export const mainPagesTitles = {
  '/': {
    breadcrumbs: [{
      title: 'System Dashboard',
      state: '/',
    }],
    headerTitle: 'System Dashboard',
  },
  'charts': {
    breadcrumbs: [{
      title: 'System Dashboard',
      state: '/charts',
    }],
    headerTitle: 'System Dashboard',
  },
  'patients': {
    breadcrumbs: [{
      title: 'Home',
      state: '/',
    }, {
      title: 'Patient Listings',
      state: '/patients',
    }],
    headerTitle: 'Patients Lists',
  },
  'profile': {
    breadcrumbs: [{
      title: 'Patient Information',
      state: '/profile',
    }],
    headerTitle: 'Personal Information',
  },
  'search-report': {
    breadcrumbs: [{
      title: 'Home',
      state: '/',
    }, {
      title: 'Search Report',
      state: '/search-report',
    }],
    headerTitle: 'System Dashboard',
  },
  'patients-summary': {
    breadcrumbs: [{
      title: 'Patient Listings',
      state: '/patients',
    }, {
      title: themeConfigs.patientsSummaryPageName,
      state: '/patients-summary',
    }],
  },
  'patients-full-details': {
    breadcrumbs: [],
    headerTitle: 'Search results',
  },
};
addPluginsPagesToLists(mainPagesTitles, [{
  title: 'Patient Listings',
  state: '/patients',
}, {
  title: themeConfigs.patientsSummaryPageName,
  state: '/patients-summary',
}]);

export const mainPagesTitlesForPatients = {
  '/': {
    breadcrumbs: [{
      title: 'Patient Summary',
      state: '/patients-summary',
    }],
    headerTitle: 'Patient Summary',
  },

  'profile': {
    breadcrumbs: [{
      title: 'Patient Information',
      state: '/profile',
    }],
    headerTitle: 'Personal Information',
  },
  'patients-summary': {
    breadcrumbs: [{
      title: themeConfigs.patientsSummaryPageName,
      state: '/patients-summary',
    }],
  },
};
addPluginsPagesToLists(mainPagesTitlesForPatients, [{
  title: themeConfigs.patientsSummaryPageName,
  state: '/patients-summary',
}]);
