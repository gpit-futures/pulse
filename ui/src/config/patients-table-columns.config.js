import React from 'react';

import { getDDMMMYYYY } from '../utils/time-helpers.utils';

const CalendarIcon = () => <i className="fa fa-calendar" />;

export const patientsColumnsConfig = [
  { key: 'name', title: 'Name', width: 150 },
  { key: 'address', title: 'Address', width: 300 },
  { key: 'dateOfBirth', title: 'Born', transformer: getDDMMMYYYY, width: 105 },
  { key: 'promsCount', title: 'PROM ', icon: <span>#</span>, width: 100 },
  { key: 'promsDate', title: 'PROM', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'gender', title: 'Gender', width: 90 },
  { key: 'id', title: 'NHS No.', width: 115 },
  { key: 'ordersDate', title: 'Orders', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'ordersCount', title: 'Orders ', icon: <span>#</span>, width: 100 },
  { key: 'resultsDate', title: 'Results', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'resultsCount', title: 'Results ', icon: <span>#</span>, width: 100 },
  { key: 'vitalsDate', title: 'Vitals', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 110 },
  { key: 'vitalsCount', title: 'Vitals ', icon: <span>#</span>, width: 100 },
  { key: 'diagnosesDate', title: 'Diagnosis', icon: <CalendarIcon />, transformer: getDDMMMYYYY, width: 130 },
  { key: 'diagnosesCount', title: 'Diagnosis ', icon: <span>#</span>, width: 120 },
  { key: 'viewPatientNavigation', title: '', icon: <span />, width: 90 },
];

export const defaultColumnsSelected = {
  name: true,
  gender: true,
  dateOfBirth: true,
  address: true,
  id: true,
  ordersDate: true,
  resultsDate: true,
  vitalsDate: false,
  diagnosesDate: false,
  ordersCount: true,
  resultsCount: true,
  vitalsCount: false,
  diagnosesCount: false,
  viewPatientNavigation: true,
  promsCount: false,
  promsDate: false,
};
