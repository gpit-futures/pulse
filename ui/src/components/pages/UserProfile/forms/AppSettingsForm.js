import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateAppSettingsForm } from '../forms.validation';
import { valuesSettingsForm, valuesSettingsFormLabels } from '../forms.config';
import { setTheme } from '../../../../ducks/set-theme.duck';

@reduxForm({
  form: 'appSettingsFormSelector',
  validate: validateAppSettingsForm,
})
export default class AppSettingsForm extends PureComponent {
  static propTypes = {
    dispatch: PropTypes.func,
    initialize: PropTypes.func,
  };

  componentDidMount() {
    const { patientsInfo, initialize } = this.props;
    initialize(this.defaultValuesForm(patientsInfo));
  }

  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesSettingsForm.APP_TITLE]: value.title,
      [valuesSettingsForm.LOGO_PATH]: value.logoB64,
      [valuesSettingsForm.SELECT_THEME]: value.themeColor,
      [valuesSettingsForm.BROWSER_TITLE]: value.browserTitle,
    };

    return defaultFormValues;
  }

 setThemeHook = dispatch => (theme) => {
   dispatch(setTheme(theme));
   return theme;
 };

 render() {
   const { dispatch } = this.props;

   return (
     <div className="panel-body-inner">
       <form name="appSettingsForm" className="form">
         <div className="form-group-wrapper">
           <Row>
             <Col xs={12} md={6}>
               <Row>
                 <Col md={11}>
                   <Field
                     label={valuesSettingsFormLabels.APP_TITLE}
                     name={valuesSettingsForm.APP_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                   <Field
                     label={valuesSettingsFormLabels.LOGO_PATH}
                     name={valuesSettingsForm.LOGO_PATH}
                     component={FileInput}
                     id={valuesSettingsForm.LOGO_PATH}
                   />
                   <Field
                     label={valuesSettingsFormLabels.SELECT_THEME}
                     name={valuesSettingsForm.SELECT_THEME}
                     component={Select}
                     options={optionsForThemesField}
                     //here we're using normalize hook for dispatching action to set app theme
                     normalize={this.setThemeHook(dispatch)}
                   />
                   <Field
                     label={valuesSettingsFormLabels.BROWSER_TITLE}
                     name={valuesSettingsForm.BROWSER_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                 </Col>
               </Row>
             </Col>
           </Row>
         </div>
       </form>
     </div>)
 }
}
