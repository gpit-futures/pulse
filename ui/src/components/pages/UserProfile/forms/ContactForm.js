import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from '../forms.validation';
import { valuesContactForm, valuesContactFormLabels } from '../forms.config';
import { defaultContactFormValues } from './default-values.config';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultContactFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="contactForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesContactFormLabels.ADDRESS}
                      name={valuesContactForm.ADDRESS}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.CITY}
                      name={valuesContactForm.CITY}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.STATE}
                      name={valuesContactForm.STATE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.POSTAL_CODE}
                      name={valuesContactForm.POSTAL_CODE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.SELECT_COUNTRY}
                      name={valuesContactForm.SELECT_COUNTRY}
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesContactFormLabels.PHONE}
                      name={valuesContactForm.PHONE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.EMAIL}
                      name={valuesContactForm.EMAIL}
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
