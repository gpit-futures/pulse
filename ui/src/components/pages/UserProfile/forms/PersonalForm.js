import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { optionsForGenderField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validatePersonalForm } from '../forms.validation';
import { valuesPersonalForm, valuesPersonalFormLabels } from '../forms.config';
import { defaultPersonalFormValues } from './default-values.config';

@reduxForm({
  form: 'personalFormSelector',
  validate: validatePersonalForm,
})
export default class PersonalForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultPersonalFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesPersonalFormLabels.FIRST_NAME}
                      name={valuesPersonalForm.FIRST_NAME}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesPersonalFormLabels.LAST_NAME}
                      name={valuesPersonalForm.LAST_NAME}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesPersonalFormLabels.NHS_NUMBER}
                      name={valuesPersonalForm.NHS_NUMBER}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesPersonalFormLabels.DATE_OF_BIRTH}
                      name={valuesPersonalForm.DATE_OF_BIRTH}
                      type="text"
                      placeholder=""
                      component={DateInput}
                      props={{ format: 'DD-MMM-YYYY' }}
                    />
                    <Field
                      label={valuesPersonalFormLabels.SELECT_GENDER}
                      name={valuesPersonalForm.SELECT_GENDER}
                      placeholder=""
                      component={Select}
                      options={optionsForGenderField}
                    />
                    <Field
                      label={valuesPersonalFormLabels.DOCTOR}
                      name={valuesPersonalForm.DOCTOR}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
