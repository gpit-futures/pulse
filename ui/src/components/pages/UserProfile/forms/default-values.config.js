import { valuesSettingsForm, valuesPersonalForm, valuesContactForm } from '../forms.config';

const defaultPersonalFormValues = {
  [valuesPersonalForm.FIRST_NAME]: 'Bob',
  [valuesPersonalForm.LAST_NAME]: 'Smith',
  [valuesPersonalForm.SELECT_GENDER]: 'female',
  [valuesPersonalForm.DOCTOR]: 'Dr Emma Huston',
  [valuesPersonalForm.DATE_OF_BIRTH]: 0,
};

const defaultContactFormValues = {
  [valuesContactForm.ADDRESS]: '6801 Tellus Street',
  [valuesContactForm.CITY]: 'Westmorland',
  [valuesContactForm.STATE]: 'Westmorland',
  [valuesContactForm.POSTAL_CODE]: 'Box 306',
  [valuesContactForm.PHONE]: '07624 647524',
  [valuesContactForm.EMAIL]: 'bob.smith@gmail.com',
};

const defaultAppSettingsFormValues = {
  [valuesSettingsForm.APP_TITLE]: 'should be fetched from server',
  [valuesSettingsForm.LOGO_PATH]: 'should be fetched from server',
  [valuesSettingsForm.SELECT_THEME]: 'themeDefault',
  [valuesSettingsForm.BROWSER_TITLE]: 'should be fetched from server',
};

export { defaultPersonalFormValues, defaultContactFormValues, defaultAppSettingsFormValues }
