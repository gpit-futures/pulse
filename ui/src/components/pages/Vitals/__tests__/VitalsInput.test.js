import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import VitalsInput from '../vitals-page-component/VitalsInput';
import { valuesNames } from '../forms.config';

Enzyme.configure({ adapter: new Adapter() });

const testProps = {
  addonName: 'test addon',
  label: 'Test label',
  placeholder: 'Test placeholder',
  input: { name: 'respirationRate', value: '' },
  meta: {
    active: false,
    error: false,
    touched: false,
  },
  id: 'test-id',
  getHighlighterClass: () => {},
  popoverLabels: 'testPopoverLabels',
  vitalStatuses: {
    [valuesNames.HEART_RATE]: {
      point: 1,
      type: 'success',
    },
    [valuesNames.LEVEL_OF_CONSCIOUSNESS]: {
      point: 0,
    },
    [valuesNames.NEWS_SCORE]: {
      point: undefined,
      type: 'success',
    },
    [valuesNames.OXYGEN_SATURATION]: {
      point: 0,
    },
    [valuesNames.OXYGEN_SUPPLEMENTAL]: {
      point: 2,
      type: 'warning',
    },
    [valuesNames.RESPIRATION_RATE]: {
      point: 1,
      type: 'success',
    },
    [valuesNames.SYSTOLIC_BP]: {
      point: 0,
    },
    [valuesNames.TEMPERATURE]: {
      point: 0,
    },
  },
  detail: {
    [valuesNames.RESPIRATION_RATE]: '12.0',
    [valuesNames.OXYGEN_SUPPLEMENTAL]: 'true',
    [valuesNames.HEART_RATE]: '45.0',
    [valuesNames.TEMPERATURE]: '37.0',
    [valuesNames.LEVEL_OF_CONSCIOUSNESS]: 'Alert',
    [valuesNames.SYSTOLIC_BP]: '112.0',
    [valuesNames.DIASTOLIC_BP]: '64.0',
    [valuesNames.OXYGEN_SATURATION]: '97.0',
    [valuesNames.NEWS_SCORE]: 3,
    [valuesNames.AUTHOR]: 'Dr Tony Shannon',
    [valuesNames.DATE_CREATED]: 1515682407000,
    [valuesNames.SOURCE]: 'ethercis',
    [valuesNames.SOURCE_ID]: '27ee5e25-4c32-46d2-b45a-f74149d72030',
  },
};
const popoverLabelsWithInputName = {
  [valuesNames.RESPIRATION_RATE]: [
    {
      place: 1,
      text: 'â‰¤ 8',
    },
    {
      place: 3,
      text: '9-11',
    },
  ],
};

describe('Component <VitalsInput />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
      <VitalsInput
        id={testProps.id}
        label={testProps.label}
        input={testProps.input}
        meta={testProps.meta}
        placeholder={testProps.placeholder}
        getHighlighterClass={testProps.getHighlighterClass}
        popoverLabels={testProps.popoverLabels}
        vitalStatuses={testProps.vitalStatuses}
        detail={testProps.detail}
        isSubmit={false}
        isNotValidate={false}
        disabled={false}
      />);

    expect(component.find('VitalsPopover')).toHaveLength(1);
    expect(component.find('input')).toHaveLength(0);
    expect(component.find('label')).toHaveLength(1);
    expect(component.find('label').text()).toEqual(testProps.label);

    expect(component.find('VitalsPopover').props().detailValue).toEqual(testProps.detail[valuesNames.RESPIRATION_RATE]);
    expect(component.find('VitalsPopover').props().disabled).toEqual(false);
    expect(component.find('VitalsPopover').props().error).toEqual(false);
    expect(component.find('VitalsPopover').props().id).toEqual(testProps.id);
    expect(component.find('VitalsPopover').props().input.name).toEqual(valuesNames.RESPIRATION_RATE);
    expect(component.find('VitalsPopover').props().isNotValidate).toEqual(false);
    expect(component.find('VitalsPopover').props().placeholder).toEqual(testProps.placeholder);
    expect(component.find('VitalsPopover').props().popoverLabels).toEqual([]);
    expect(component.find('VitalsPopover').props().showError).toEqual(false);
    expect(component.find('VitalsPopover').props().title).toEqual(testProps.label);

    expect(component).toMatchSnapshot();
  });

  it('should renders with all props correctly when showError', () => {
    const component = shallow(
      <VitalsInput
        id={testProps.id}
        label={testProps.label}
        input={testProps.input}
        meta={{ active: false, error: true }}
        placeholder={testProps.placeholder}
        getHighlighterClass={testProps.getHighlighterClass}
        popoverLabels={popoverLabelsWithInputName}
        vitalStatuses={{}}
        detail={testProps.detail}
        isSubmit={true}
        isNotValidate={false}
        disabled={false}
      />);

    expect(component.find('VitalsPopover')).toHaveLength(1);
    expect(component.find('input')).toHaveLength(0);
    expect(component.find('label')).toHaveLength(1);
    expect(component.find('label').text()).toEqual(testProps.label);
    expect(component.find('.has-error')).toHaveLength(1);
    expect(component.find('.vitals-group').props().className).toEqual('vitals-group highlighter-wrapper has-error');

    expect(component).toMatchSnapshot();
  });

  it('should renders with all props correctly when not showError', () => {
    const component = shallow(
      <VitalsInput
        id={testProps.id}
        label={testProps.label}
        input={testProps.input}
        meta={{ active: false, error: false }}
        placeholder={testProps.placeholder}
        getHighlighterClass={testProps.getHighlighterClass}
        popoverLabels={popoverLabelsWithInputName}
        vitalStatuses={{}}
        detail={testProps.detail}
        isSubmit={false}
        isNotValidate={false}
        disabled={false}
      />);

    expect(component.state().isChanged).toEqual(false);
    component.setProps({ meta: { dirty: true }});
    expect(component.state().isChanged).toEqual(true);

    expect(component.find('VitalsPopover')).toHaveLength(1);
    expect(component.find('input')).toHaveLength(0);
    expect(component.find('label')).toHaveLength(1);
    expect(component.find('label').text()).toEqual(testProps.label);
    expect(component.find('.has-success')).toHaveLength(1);
    expect(component.find('.vitals-group').props().className).toEqual('vitals-group highlighter-wrapper has-success');

    expect(component).toMatchSnapshot();

    component.setProps({ meta: { dirty: false }});
  });


  it('should renders with all props correctly testing without popover', () => {
    const component = shallow(
      <VitalsInput
        id={testProps.id}
        label={testProps.label}
        input={testProps.input}
        meta={testProps.meta}
        placeholder={testProps.placeholder}
        getHighlighterClass={testProps.getHighlighterClass}
        popoverLabels={testProps.popoverLabels}
        vitalStatuses={testProps.vitalStatuses}
        detail={testProps.detail}
        isSubmit={false}
        isNotValidate={false}
        disabled={false}
        withoutPopover
      />);

    expect(component.find('VitalsPopover')).toHaveLength(0);
    expect(component.find('input')).toHaveLength(1);
    expect(component.find('label')).toHaveLength(1);
    expect(component.find('label').text()).toEqual(testProps.label);
    expect(component.find('.input-holder').props().className).toEqual('input-holder vitals-holder success');
    expect(component.find('.success')).toHaveLength(1);
    expect(component.find('input').props().disabled).toEqual(false);
    expect(component.find('input').props().id).toEqual(testProps.id);
    expect(component.find('input').props().name).toEqual(valuesNames.RESPIRATION_RATE);
    expect(component.find('input').props().placeholder).toEqual(testProps.placeholder);

    expect(component).toMatchSnapshot();
  });
});
