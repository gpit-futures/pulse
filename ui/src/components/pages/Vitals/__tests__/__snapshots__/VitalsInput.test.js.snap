// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <VitalsInput /> should renders with all props correctly 1`] = `
<div
  className="vitals-group highlighter-wrapper"
>
  <span
    className="undefined"
  />
  <label
    className="vitals-label"
  >
    Test label
  </label>
  <VitalsPopover
    detailValue="12.0"
    disabled={false}
    error={false}
    id="test-id"
    input={
      Object {
        "name": "respirationRate",
        "value": "",
      }
    }
    isNotValidate={false}
    placeholder="Test placeholder"
    popoverLabels={Array []}
    showError={false}
    title="Test label"
    vitalStatusesType="success"
  />
</div>
`;

exports[`Component <VitalsInput /> should renders with all props correctly testing without popover 1`] = `
<div
  className="vitals-group highlighter-wrapper"
>
  <span
    className="undefined"
  />
  <label
    className="vitals-label"
  >
    Test label
  </label>
  <div
    className="input-holder vitals-holder success"
  >
    <input
      className="form-control input-sm"
      disabled={false}
      id="test-id"
      name="respirationRate"
      placeholder="Test placeholder"
      value=""
    />
  </div>
</div>
`;

exports[`Component <VitalsInput /> should renders with all props correctly when not showError 1`] = `
<div
  className="vitals-group highlighter-wrapper has-success"
>
  <span
    className="undefined"
  />
  <label
    className="vitals-label"
  >
    Test label
  </label>
  <VitalsPopover
    detailValue="12.0"
    disabled={false}
    id="test-id"
    input={
      Object {
        "name": "respirationRate",
        "value": "",
      }
    }
    isNotValidate={false}
    placeholder="Test placeholder"
    popoverLabels={
      Array [
        Object {
          "place": 1,
          "text": "≤ 8",
        },
        Object {
          "place": 3,
          "text": "9-11",
        },
      ]
    }
    title="Test label"
    vitalStatusesType=""
  />
</div>
`;

exports[`Component <VitalsInput /> should renders with all props correctly when showError 1`] = `
<div
  className="vitals-group highlighter-wrapper has-error"
>
  <span
    className="undefined"
  />
  <label
    className="vitals-label"
  >
    Test label
  </label>
  <VitalsPopover
    detailValue="12.0"
    disabled={false}
    error={true}
    id="test-id"
    input={
      Object {
        "name": "respirationRate",
        "value": "",
      }
    }
    isNotValidate={false}
    placeholder="Test placeholder"
    popoverLabels={
      Array [
        Object {
          "place": 1,
          "text": "≤ 8",
        },
        Object {
          "place": 3,
          "text": "9-11",
        },
      ]
    }
    showError={true}
    title="Test label"
    vitalStatusesType=""
  />
</div>
`;
